basiclibrary/src/main/java/basiclibrary/Library.java

```diff
@@ -67,8 +70,88 @@ public static double calculatingAverage(int[] testAverageArr) {
 
+    //Data: Daily average temperatures for Seattle, October 1-28 2017
+    //int[][] weeklyMonthTemperatures = {
+    //  {66, 64, 58, 65, 71, 57, 60},
+    //  {57, 65, 65, 70, 72, 65, 51},
+    //  {55, 54, 60, 53, 59, 57, 61},
+    //  {65, 56, 55, 52, 55, 62, 57}
+    //};
 
+    //Expected output:
+    //High: 72
+    //Low: 51
+    //Never saw temperature: 63
+    //Never saw temperature: 67
+    //Never saw temperature: 68
+    //Never saw temperature: 69
+
+    //Iterate through the data to find the min and max values
+    //Use an int HashSet to track all unique temperatures in the data set
+    //Iterate from the min to max temp and create a String containing any temperature not seen during the month
+
+    public static String analyzeWeatherData(int[][] weeklyMonthTemperatures){
+        int minTemperature = Integer.MAX_VALUE;
+        int maxTemperature = Integer.MIN_VALUE;
+        String returnDataString = "";
+
+        HashSet<Integer> trackWeatherData = new HashSet<>();
+
+        for (int i = 0; i < weeklyMonthTemperatures.length; i++) {
+            for (int j = 0; j < weeklyMonthTemperatures[i].length; j++) {
+                trackWeatherData.add(weeklyMonthTemperatures[i][j]);
+                if ( weeklyMonthTemperatures[i][j] < minTemperature) {
+                    minTemperature = weeklyMonthTemperatures[i][j];
+                }
+                if ( weeklyMonthTemperatures[i][j] > maxTemperature) {
+                    maxTemperature = weeklyMonthTemperatures[i][j];
+                }
+            }
+        }
+
+        returnDataString += "High: " + maxTemperature + "\n";
+        returnDataString += "Low: " + minTemperature + "\n";
+
+        for (int i = minTemperature + 1; i < maxTemperature; i++) {
+            if (!trackWeatherData.contains(i)) {
+                returnDataString += "Never saw temperature: " + i+ "\n";
+            }
+        }
+        return returnDataString;
+    }
+
+    //Tallying Election: Write a function called tally that accepts a List of Strings representing votes and returns one string to show what got the most votes.
+    //Within main method include
+        //List<String> votes = new ArrayList<>();
+            //votes.add("Bush");
+            //votes.add("Bush");
+            //votes.add("Bush");
+            //votes.add("Shrub");
+            //votes.add("Hedge");
+            //votes.add("Shrub");
+            //votes.add("Bush");
+            //votes.add("Hedge");
+            //votes.add("Bush");
+        //String winner = tally(votes);
+        //System.out.println(winner + " received the most votes!");
+
+    public static String tally(List<String> votesList) {
+        votesList = new ArrayList<>();
```

Comment:
as of now, we are not using a data structure that stores counts easily. I would recommend creating a new HashMap instead where you store the candidate as the key and the count of votes as the value.